enterprise/coderd/portsharing/portsharing.go

```diff
@@ -0,0 +1,30 @@
+package portsharing
+
+import (
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+type EnterprisePortSharer struct{}
+
+func NewEnterprisePortSharer() *EnterprisePortSharer {
+	return &EnterprisePortSharer{}
+}
+
+func (EnterprisePortSharer) AuthorizedPortSharingLevel(template database.Template, level codersdk.WorkspaceAgentPortShareLevel) error {
+	if level > codersdk.WorkspaceAgentPortShareLevel(template.MaxPortSharingLevel) {
+		return xerrors.Errorf("port sharing level not allowed. Must not be greater than '%s'", template.MaxPortSharingLevel)
+	}
```

Comment:
You're comparing two strings with `>`. You should instead use a switch or something.